lib/scope.js

```diff
@@ -110,7 +110,11 @@ AST_Toplevel.DEFMETHOD("figure_out_scope", function(options){
 
     // pass 1: setup scope chaining and handle definitions
     var self = this;
-    var scope = self.parent_scope = null;
+    var scope = null;
+    if (options.module) {
+        scope = new AST_Scope(self);
+    }
+    self.parent_scope = scope;
```

Comment:
Rather than creating a new scope all that needs to be done is to set `mangle` `toplevel` to `true` when `mangle` `module` is enabled. Then it will be handled automatically.